shark_turbine/kernel/wave/minimize_global_loads.py

```diff
@@ -140,9 +141,7 @@ def add_optimized_nodes(
                         write = Write(
                             read, custom_user.memory, load_elems_per_thread
                         ).add_to_graph(custom.graph)
-                        write.index = remove_global_indexing(
-                            read.index, tilingConstraints
-                        )
+                        write.index = deepcopy(read.index)
```

Comment:
Why do we need `deepcopy` now?

Reply:
So if we just assign read.index to write.index, then when we modify write.index in the corrections pass it will also affect read.index. But did you have a better way to handle this?

Reply:
But `remove_global_indexing` is already creating a copy, it shouldn't affect original index object.

Reply:
Yes, that's right. `remove_global_indexing` is creating a copy, but that copy will be assigned to write.index and read.index which is not what we want. If you remove the `deepcopy`, two of the lit tests fail.

Reply:
okay I was able to figure this out. I have removed the deep copy. @Hardcode84 - can you take a look when you get a chance? thanks!